bunsen/bunsen-core/src/main/java/com/cerner/bunsen/definitions/StructureDefinitions.java

```diff
@@ -545,12 +553,24 @@ private <T> T transform(
       List<String> referenceProfiles = parentElement.getReferenceTargetProfiles();
```

Comment:
Looks like special handling for `Reference` types is still needed as we add all the `references` and the `children` to the converter as done over [here](https://github.com/google/fhir-data-pipes/blob/498d7c1b336bf77915fc39ca973b4231b594bf28/bunsen/bunsen-avro/src/main/java/com/cerner/bunsen/avro/converters/DefinitionToAvroVisitor.java#L712) and [here](https://github.com/google/fhir-data-pipes/blob/498d7c1b336bf77915fc39ca973b4231b594bf28/bunsen/bunsen-avro/src/main/java/com/cerner/bunsen/avro/converters/DefinitionToAvroVisitor.java#L735). That said, this has been the code since beginning. The fix that was done to this branch was over [here](https://github.com/google/fhir-data-pipes/pull/634/files#diff-3074f63a61ae44ee27049eccc58e6ae935ffe5a35e159de6c533cb751a6b1fe7), where a unique name was supposed to be generated but it was ending up with `Identifier.assignerReference` irrespective of the stack path. This has been fixed in the current PR, where the name depends on the path from the root in the format `<PathFromRoot>+<ElementName>Reference`  e.g. `PatientReferenceIdentifierAssignerReference` will be name for `Patient.generalPractitioner.identifier.assigner` and `Patient.managingOrganization.identifier.assigner`. However the names for `Patient.generalPractitioner` and `Patient.managingOrganization` are different i.e. `PatientGeneralPractitionerReference` and `PatientManagingOrganizationReference` respectively. This makes sense because the `Patient.generalPractitioner` references multiple reference types and should be different from `Patient.managingOrganization`.

Reply:
I see, thanks for checking and the clarification. Yeah, we need to investigate more that increase in schema size and take action if necessary (after this PR).

Reply:
sure